questions.md

```diff
@@ -2,29 +2,72 @@
 
 ## HPC good citizenship
 
-TESTING...TESTING
 1. On the UCI cluster, the resource request "-pe openmp 64" refers to the number of processors requested.  Does that
    request mean that your commands will use multiple processors?
+
+answer: no some commands do not use multiple processors. To note: some commands/programs allow you to choose the number of processors that are used.  So the program would use the number of processors chosen by the user in the command line.  So if you request 64 processors "-pe openmp 64", but only declare 20 in the command line script the command will use 20 processors.  
+
 2. In general, how do you know how many processors, how much RAM, how many files would be required/needed/written by the
    jobs you are running on the cluster?
+
+answer: Before running a program you can determine the ballpark RAM, # of files, etc. by reading the a program manual (if there is one) or googling the info online for a particular command.  However, this info many times is not available online.  Before running a command or program for the first time using your own data you should use a smaller test data set and run a prefix command before the program/command you want to run. You will most likely have to scale up the amount of RAM needed to match the size of your dataset. The prefix command is:
+
+```bash
+/usr/bin/time -v
+```  
+
 3. In order to be a "good citizen", you need to have some idea of much RAM your job requires.  In particular, you need
    to know the "peak" (i.e., maximum) RAM required at any point during execution.  Show an example of the shell command
    that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak RAM values are written to a file.
-4. What are the units of your answer for number 3?
+answer:
+The shell command I would use would be:
+```bash
+/usr/bin/time -o stats.txt my_program
```

Comment:
Hi, I ran this and did get an answer, but it was a bit difficult to parse. I recommend using the same command above. env time -v -o output command